contracts/router/modules/pool/velodrome/VelodromeV2Module.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {IERC20, SafeERC20} from "@openzeppelin/contracts-4.5.0/token/ERC20/utils/SafeERC20.sol";
+
+import {IndexedToken, IPoolModule} from "../../../interfaces/IPoolModule.sol";
+
+import {IVelodromeV2Pool} from "../../../interfaces/velodrome/IVelodromeV2Pool.sol";
+import {IVelodromeV2Router} from "../../../interfaces/velodrome/IVelodromeV2Router.sol";
+
+/// @notice PoolModule for Velodrome V2 pools
+/// @dev Implements IPoolModule interface to be used with pools added to LinkedPool router
+contract VelodromeV2Module is IPoolModule {
+    using SafeERC20 for IERC20;
+
+    IVelodromeV2Router public immutable router;
+
+    constructor(address _router) {
+        router = IVelodromeV2Router(_router);
+    }
+
+    function poolSwap(
+        address pool,
+        IndexedToken memory tokenFrom,
+        IndexedToken memory tokenTo,
+        uint256 amountIn
+    ) external returns (uint256 amountOut) {
+        // TODO: sanity checks on tokenFrom, tokenTo indices <= 1?
+        address tokenIn = tokenFrom.token;
+        IERC20(tokenIn).safeApprove(address(router), amountIn);
+
+        bool stable = IVelodromeV2Pool(pool).stable();
+        address factory = IVelodromeV2Pool(pool).factory();
+        IVelodromeV2Router.Route[] memory routes = new IVelodromeV2Router.Route[](1);
+        routes[0] = IVelodromeV2Router.Route({
+            from: tokenFrom.token,
+            to: tokenTo.token,
+            stable: stable,
+            factory: factory
+        });
+
+        uint256[] memory amounts = new uint256[](2);
+        amounts = router.swapExactTokensForTokens(amountIn, 0, routes, address(this), block.timestamp);
+        amountOut = amounts[1];
+    }
+
+    /// @dev Careful with pool.getAmountOut as will return token1 as in if give invalid tokenIn address (token not in pool)
+    function getPoolQuote(
+        address pool,
+        IndexedToken memory tokenFrom,
+        IndexedToken memory tokenTo,
+        uint256 amountIn,
+        bool probePaused
+    ) external view returns (uint256 amountOut) {
+        address tokenIn = tokenFrom.token;
```

Comment:
I think checking `(tokenFrom, tokenTo)` against `pool.tokens()` makes sense here. Indexes don't really need to be checked, as they are not going to be later used anyway.